algorithms/tendermint/verifier/src/lib.rs

```diff
@@ -0,0 +1,48 @@
+//! This section mainly has been ported from `InformalSystems/hermes/relayer/src/light_client`
+use futures_util::future::TryFutureExt;
+use ibc::Height as ICSHeight;
+use ics07_tendermint::client_state::ClientState as TmClientState;
+use pallet_ibc::light_clients::HostFunctionsManager;
+use tendermint_light_client::{
+	components::{self, io::AsyncIo, io::AtHeight, io::RpcIo},
+	light_client::LightClient as TmLightClient,
+	state::State as LightClientState,
+	store::{memory::MemoryStore, LightStore},
+};
+use tendermint_light_client_verifier::{
+	host_functions::CryptoProvider,
+	operations::{ProdCommitValidator, ProdVotingPowerCalculator},
+	options::Options as TmOptions,
+	predicates::ProdPredicates,
+	types::{Height as TMHeight, LightBlock, PeerId, Status},
+	PredicateVerifier, ProdVerifier,
+};
+use tendermint_primitives::error::Error;
+use tendermint_rpc::{Client, HttpClient, Url};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Verified<H> {
+	/// Verified target
+	pub target: H,
+	/// Supporting headers needed to verify `target`
+	pub supporting: Vec<H>,
+}
+
+#[derive(Clone, Debug)]
+pub struct LightClient {
+	peer_id: PeerId,
+	io: RpcIo,
```

Comment:
`io` also is a bit generic name. I'd recommend `rpc_handler` or something along those lines. What do you think?